sql-plugin/src/main/scala/com/nvidia/spark/rapids/aggregate.scala

```diff
@@ -852,13 +852,39 @@ abstract class GpuBaseAggregateMeta[INPUT <: SparkPlan](
     groupingExpressions ++ aggregateExpressions ++ aggregateAttributes ++ resultExpressions
 
   override def tagPlanForGpu(): Unit = {
-    // We don't support Arrays and Maps as GroupBy keys yet, even they are nested in Structs. So,
+    // We don't support Maps as GroupBy keys yet, even if they are nested in Structs. So,
     // we need to run recursive type check on the structs.
-    val arrayOrMapGroupings = agg.groupingExpressions.exists(e =>
+    val mapGroupings = agg.groupingExpressions.exists(e =>
       TrampolineUtil.dataTypeExistsRecursively(e.dataType,
-        dt => dt.isInstanceOf[ArrayType] || dt.isInstanceOf[MapType]))
-    if (arrayOrMapGroupings) {
-      willNotWorkOnGpu("ArrayTypes or MapTypes in grouping expressions are not supported")
+        dt => dt.isInstanceOf[MapType]))
+    if (mapGroupings) {
+      willNotWorkOnGpu("MapTypes in grouping expressions are not supported")
+    }
+
+    def getArrayTypesRecursively(dt: DataType): List[DataType] = {
+      // dt shouldn't have any maps at this stage so, we aren't checking for it
+      dt match {
+        case StructType(fields) =>
+          fields.flatMap(f => getArrayTypesRecursively(f.dataType)).toList
+        case _ =>
+          if (dt.isInstanceOf[ArrayType]) {
+            List(dt)
```

Comment:
Function name implies this will truly recurse, but it doesn't recurse on array types.  Either that's a bug or there should be a comment explaining the recurse exception and why.